home/.bin/create-hotfix

```diff
@@ -20,47 +26,61 @@ require 'active_support'
 # i.e.
 #    export PIVOTAL_TOKEN="TOKEN HERE"
 #    export PIVOTAL_PROJECT_ID="some_number_here"
-#
+
 #    You can find your token at the bottom of the User Profile page on Pivotal,
 #    which you can get to by clicking your name in the top right corner.
-#
+
 # How you'd use it
 #     i.e. create-hotfix 'Do The Needful'
-#
-# What it does
-#
-# It will detect who you are hitched with and add both as story owners. It won't
-# run if you have unstaged git changes. It will then create the branch name using
-# the slugify pivotal card bookmarklet logic.
-#
-####
 
 # trolling
 class GitError < StandardError; end
 
-#check for uncommitted git files because git-flow will blow up
-uncommitted_files = `git status --porcelain`
+def master_up_to_date?(repo)
+  branch_up_to_date?(repo, 'master')
+end
+
+def develop_up_to_date?(repo)
+  branch_up_to_date?(repo, 'develop')
+end
+
+def branch_up_to_date?(repo, branch_name)
+  repo.revparse(branch_name) == repo.revparse("origin/#{branch_name}")
+end
 
-if !uncommitted_files.empty?
-  raise GitError, "you have uncommitted changes"
+def env_variable_undefined?(variable)
+  variable.nil? || variable.empty?
 end
 
+repository_dir = Dir.pwd
```

Comment:
Fixed! Tom helped me come up with a solution where the script will recursively go up until it finds the .git file. It will blow up immediately if there's no chance of finding a .git file from where you are.